10-JavaScript/closures-continued.md

```diff
@@ -4,48 +4,110 @@
 
 **A closure is a way to attach some extra information to a function.**
 
-## Closures with jQuery
 
-Now that we've explored the concepts behind closures a bit, let's return to our [original example with the buttons](https://codepen.io/adadev/pen/ZmbMdq).
+Doing event handling in the browser is something new and we will handle it a bit differently in React.  However these are common use-cases where people take advantage of closures when interacting with the DOM.
+
+## Closures with vanilla (plain) JavaScript
+
+Now that we've explored the concepts behind closures a bit, let's return to our [original example with the buttons](https://codesandbox.io/s/kind-ritchie-6z3dz).
 
 With your neighbors, look at the JavaScript code in that pen, open your Dev Tools to see the console, and answer the following questions:
 
+- Where is the closure created?
 - What is the outer function?
 - What is the inner function?
 - What variable from the outer function is being referenced in the inner function?
 - How many different click handlers are created by this code?
 
+## JavaScript in the browser
+
+### Where the code starts running
+
+The code below tells Javascript to call `mainFunction` after the HTML page has finished loading.
+
+```javascript
+// Run the main function only after the HML has loaded
+if (document.readyState === "loading") {
+  // Loading hasn't finished yet
+  document.addEventListener("DOMContentLoaded", mainFunction);
+} else {
+  // `DOMContentLoaded` has already fired
+  mainFunction();
+}
+```
+
+### The mainFunction
+
+The `mainFunction` takes a list of colors and loops through them.  It finds an HTML element with the CSS selector `button.${color}`.  Then it creates a click handler for that color (more later).  Lastly it tells the browser that when that element is clicked (the click event) run the `clickHandler` function.
+
+```javascript
+const mainFunction = () => {
+  ["red", "blue", "green"].forEach(color => {
+    const element = document.querySelector(`button.${color}`);
+
+    const clickHandler = buildClickHandler(color);
+
+    element.addEventListener("click", clickHandler);
+  });
+};
+```
+
+### The buildClickHandler
+
+The `buildClickHandler` function takes a color as a parameter and returns an **inner function** which:
+
+1.  Finds the HTML element with the id `square`.
+2.  Removes any class on that element
+3.  Sets the class to be the same name as the color.
+
+So this `buildClickHandler` creates a closure when called.  The extra information the returned function provides is the color to make the square.
+
 ## Exercise: Reading Closures
 
-Take a look at the following codepens:
+Take a look at the following code sanboxes:
 
-- [Adder](https://codepen.io/adadev/pen/gXXaQm?editors=0010)
-- [Stopwatch](https://codepen.io/adadev/pen/GwpXmJ?editors=1010)
-- [Helpful Hints](https://codepen.io/adadev/pen/EbbVqq?editors=1010)
+- [Adder](https://codesandbox.io/s/makeadder-98hn4)
```

Comment:
This was just a copy-paste error; since Adder was in the intro to Closures, it's probably okay to take it out of this list.

Reply:
It was in the previous intro to closures so... I'd left it in.